helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -158,7 +162,7 @@ private void joinCluster() {
     }
   }
 
-  private void createLiveInstance() {
+  private void createLiveInstance(final String sessionId) {
```

Comment:
I don't think so. If you pass the _sessionId as a parameter, even it is changed later at random time by the manager logic, the parameter won't be changed. Note that you are passing the String.

Reply:
Discussed offline, it seems my purpose will leave a gap which might cause a race condition. It seems that we have to add this new interface.

Reply:
My 2 cents from our discussion offline.

Since we've had some issues with keeping the sessionId as an internal state (like _sessionId as a private variable), I prefer to take it in as the method parameter to do away with any state. This way, we don't treat sessionId as an internal state and its lifecycle will coincide with that of the method.